tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute.Exceptions;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class ManagedIdentityTests
+    {
+        private static readonly string[] s_msi_scopes = { "https://management.azure.com" };
+        private static readonly string s_clientId = "client_id";
+
+        //http proxy base URL 
+        private static readonly string s_baseURL = "https://service.msidlab.com/";
+
+        //Shared User Assigned Client ID
+        private const string UserAssignedClientID = "3b57c42c-3201-4295-ae27-d6baec5b7027";
+        
+        //Resource ID of the User Assigned Identity 
+        private const string Mi_res_id = "/subscriptions/c1686c51-b717-4fe0-9af3-24a20a41fb0c/" +
+            "resourcegroups/MSAL_MSI/providers/Microsoft.ManagedIdentity/userAssignedIdentities/" +
+            "MSAL_MSI_USERID";
+
+        [TestMethod]
+        public async Task ManagedIdentitySourceCheckAsync()
+        {
+            //Arrange
+            string result = string.Empty;
+            string expectedClientException = "[Managed Identity] Authentication unavailable. " +
+                "No response received from the managed identity endpoint.";
+
+            IConfidentialClientApplication cca = CreateCCAWithProxy(s_baseURL);
+
+            //Act
+            try
+            {
+                AuthenticationResult authenticationResult = await cca
+                    .AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity()
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+            }
+            catch (MsalServiceException ex)
+            {
+                result = ex.Message;
+            }
+
+            //Assert
+            Assert.AreSame(result, expectedClientException);
```

Comment:
Use [Assert.ThrowsException](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8e21adc1c5f3c4440044f07b0edcdcc25fe0ba2a/tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/AppServiceTests.cs#L95)